### DISCUSSION TITLE
Quarkus at Devoxx Belgium

### DISCUSSION THREAD

**Discussion by insectengine** [quarkusio/quarkus#35361] [Events]
**Event Description:** At Devoxx Belgium 2023, attendees can look forward to a spectrum of talks and sessions from the industry’s brightest minds, spotlighting both local and global talent.

**Date:** October 2-6, 2023
**Location:** Antwerp, Belgium
**Event Type:** In Person

https://devoxx.be/

****
### Quarkus Sessions:

**MONDAY**

**Taming performance issues into the wild: a practical guide to JVM profiling**
**[Watch the Session on Youtube](https://www.youtube.com/watch?v=Cw4nN5L-2vU&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=4&t=7s&pp=iAQB)**
**Session Type:** DEEP DIVE
**Speaker(s):** Francesco Nigro & Mario Fusco
**Day/Time:** Monday from 13:30 - 16:30
**Abstract:** The session will start with a quick introduction to the theory of profiling discussing the motivations, explaining the different types of profiling and visualization format while listing the tools available for this purpose. This also includes some tooling for reliably emulating the load generation and validating the improvements.

Then we will guide the attendees through the profiling tools that we want to use throughout the session:

- Java VisualVM: https://visualvm.github.io/download.html
- Async-profiler: https://github.com/jvm-profiling-tools/async-profiler
- JDK Mission Control: https://www.oracle.com/java/technologies/jdk-mission-control.html
- JMH: https://github.com/openjdk/jmh
- Hyperfoil: https://hyperfoil.io/ (for load generation)

Then we will put these tools at work in a practical real-world scenario. We will provide a sample Quarkus-based Java webapp, using a simple but realistic technology stack, having different performance issues involving excessive memory allocation, CPU consumption, slow blocking I/O, locks contention and cache misses. We will demonstrate how to use the before mentioned profiling tools to discover, investigate, fix and verify these issues.

**Debugging in the Cloud, Mastered Locally with JKube remote-dev**
**[Watch the Session on Youtube](https://www.youtube.com/watch?v=feL1uL-PLr8&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=15&pp=iAQB)**
**Session Type:** TOOLS-IN-ACTION 
**Speaker(s):** Sun Tan
**Day/Time:** Monday from 16:50 - 17:20
**Abstract:**  Ever found yourself debugging your application in a live production environment? Not recommended, right? But fear not! I understand the struggle. Reproducing bugs on your laptop can be a challenge when you rely on various interconnected services like databases or authentication services.

That's where Kubernetes and Infrastructure as Code come to the rescue. It offers the ability to replicate a complete production environment within the confines of a controlled developer sandbox. But even with its powers, the ease of debugging and making live modifications may not be as seamless as one would hope on a familiar laptop.

During this presentation, I will introduce Eclipse JKube's Maven/Gradle plugins, which allow you to effortlessly deploy your Java apps to Kubernetes. And the cherry on top? The remarkable remote-dev feature. With it, you can live-code, debug, and test your app with all its dependencies, just like in the cloud, but right on your local machine. Don't miss out on live demos showcasing this magic!

****
**TUESDAY**

**[Put a versatile memory grid in your Quarkus project with Apache Ignite](https://devoxx.be/talk/?id=50754)**
**Session Type:** HANDS-ON LAB
**Speaker(s):** Antoine Sabot-Durand, Pierre Cesmat, and Yann Blazart
**Day/Time:** Tuesday from 09:30 - 12:30
**Abstract:** Imagine a distributed memory database that in addition to the basic key/value storage gives you SQL engine, a proxy to your classic db to cache data, a small message broker, a computation tool for ML model, memory to disk persistence and a lot more ! You have the beginning of the amazing Apache Ignite features list.

Mix this powerful tool with Quarkus, the nextgen microservices framework and you’ll obtain a unique combination to interact with your data.

In this lab, we’ll show you how to get started with Ignite in a Quarkus environment. After having setup your ignite cluster, you’ll create a bunch of microservices to exploit Ignite powerful distributed computation.

The only requirement for this lab is an IDE, Java 17+ and Maven 3.9+

**[Building Cloud Native Applications with Quarkus, GraalVM and some AI](https://devoxx.be/talk/?id=20297)** 
**Session Type:** HANDS-ON LAB
**Speaker(s):** Antonio Goncalves, Holly Cummins, and Georgios Andrianakis
**Day/Time:**  Tuesday from 13:30 - 16:30
**Abstract:** Cloud-native applications in Java are great, but… despite a fantastic ecosystem around Java, it falls short in containers, in comparison to Go or Node.js. The initial memory consumption, the startup time, and the optimizations for long-running processes are killer criteria in the cloud. Quarkus is a Kubernetes native Java stack tailored for GraalVM and OpenJDK. With Quarkus you can compile your application into a native executable, reduce the startup time to a few milliseconds, reduce the memory consumption to a few MB, and make applications ideally suited to run in containers. And with just a few lines of code Quarkus can invoke OpenAI to become a storyteller.

This workshop explores why it matters and how you can leverage Quarkus and GraalVM to build supersonic, subatomic, cloud-native applications.

In this workshop, you will learn how to:

- Build Web applications with Quarkus
- Use Hibernate and Hibernate with Panache
- Build event-driven microservices with Apache Kafka
- Monitor your applications
- Prompt OpenAI with Semantic Kernel APIs

This is a hands-on experience, so please bring your own laptop with the following installed and configured on your laptop prior to the start:

- Java 17
- A working container runtime environment (Docker, Podman, Rancher Desktop, etc)
- GraalVM 22.3, Java 17 version

1. Linux - https://www.graalvm.org/docs/getting-started/linux
2. Windows - https://www.graalvm.org/docs/getting-started/windows
3. MacOS - https://www.graalvm.org/docs/getting-started/macos

- An OpenAI account https://platform.openai.com

**** 
**WEDNESDAY**

**10 Years of The Well-Grounded Java Developer** 
**[Watch the Session on YouTube](https://www.youtube.com/watch?v=4mY-MFmSO2c&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=180)**
**Session Type:** CONFERENCE
**Speaker(s):** Ben Evans 
**Day/Time:**  Wednesday from 15:10 - 16:00
**Abstract:**  The first edition of "The Well-Grounded Java Developer" was published 10 years ago - back when Java 7 was the new kid in town.

In this talk, Ben Evans explores the themes that underwrote the first edition and looks back at how things have changed (and what's stayed the same) and what went into the recently published second edition. We will explore the question of what it means to be "Well-Grounded" - the skills and the curiosity that the term implies, and what that means for the modern developer.
We'll also look to the future and see what lies ahead -- what does an aspiring Java technologist need to be keeping an eye on as they seek to stay Well-Grounded in a changing world of software.

**Asynchronous Ruby on the JVM with Project Loom**
**[Watch the Session on YouTube](https://www.youtube.com/watch?v=1zSF1259s6w&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=28)**
**Session Type:** CONFERENCE 
**Speaker(s):**  Charles Nutter
**Day/Time:**  Wednesday from 17:50 - 18:40
**Abstract:**  Project Loom's virtual threads are one of the most exciting additions to the JVM in history, and JRuby provides a powerful way to take advantage of them using Ruby's Fiber subsystem and gems like async and falcon. This talk will show how JRuby has integrated virtual threads to support lightweight fibers, and showcase the libraries and frameworks from the Ruby world that leverage fibers to efficiently handle thousands of concurrent operations. We'll do a deep dive on the performance of fibers with native versus virtual threads, compare scaling high-concurrency applications on JRuby versus the standard C-based Ruby implementation, and show how you can take advantage of virtual threads today, with or without Ruby.

**** 
**THURSDAY**

**Quarkus 3: The Road to Loom for Cheaper, Faster, and Easier Concurrent Applications**
**[Watch the Session on Youtube](https://www.youtube.com/watch?v=sl4UGdKvC3o&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=113&pp=iAQB)**
**Session Type:** CONFERENCE 
**Speaker(s):** Daniel Oh
**Day/Time:** Thursday from 09:30 - 10:20
**Abstract:** Developers have had extraordinary experiences and taken stunning capabilities while they were developing Java microservices from local to the cloud using Quarkus for the past 4 years. But there is more to come with their journey since we have just arrived in Quarkus 3 which provides good, better, and even excellent features and capabilities in terms of developer experience, performance, scalability, and cloud integration. Especially, Quarkus 3 simplifies asynchronous concurrent applications using virtual threads (Project Loom) for high scalability. 

In this session, we take you through how Quarkus integrates Loom for developers to make concurrent applications easier, have cheaper memory, and have high performance using virtual threads. You can also learn about what’s new in Quarkus 3 such as JakartaEE 10, MicroProfile 6, Hibernate ORM 6, and more.

**Tradeoffs, Bad Science, and Polar Bears – The World of Java Optimisation** 
**[Watch the Session on YouTube](https://www.youtube.com/watch?v=PvZaDo_I21g&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=125&pp=iAQB)**
**Session Type:** CONFERENCE 
**Speaker(s):** Holly Cummins
**Day/Time:** Thursday from 9:30 - 10:20
**Abstract:** Welcome to the Java optimisation jungle. Why can’t we “just make it go faster”? It turns out, in most cases, we need to first work out “faster for whom?” and “why do we want to go faster?” and “what even is faster?” This talk introduces the basic principles of optimisation, before bouncing through the pitfalls of optimisation; why the exact same techniques which make Quarkus rocket-fast used to be a terrible idea fifteen years ago, why fast benchmarks make for slow programs, why project loom may not be the speedup you're hoping for, and why even though it can be easy to get wrong, optimisation really really matters. Along the way we’ll talk about measuring things, bad advice, garbage collection, and climate change.

**Kubernetes. From 0 to Production in Java**
**[Watch the Session on YouTube](https://www.youtube.com/watch?v=EB8kdbAbSmE&list=PLRsbF2sD7JVoSl0q9XTZ_RKlucdQL5sBp&index=105&pp=iAQB)**
**Session Type:** CONFERENCE 
**Speaker(s):** Kevin Dubois & Alex Soto
**Day/Time:** Thursday from 13:50 - 14:40
**Abstract:** I hear you thinking, “is this yet another session about doing cloud native development?”. Well.. yes, sort of, but we promise it’s going to be a good one - or so we think :). 
 
In this session we’re going to talk (and demo!) about what it means to be a good Cloud Citizen - which is more than merely putting an application in a container and deploying it. To optimize your (Java) application's performance and resilience in the cloud, it is essential to consider factors such as providing real-time health status through fine-grained metrics. You’ll also need to ensure fast startup, and avoid excessive resource consumption within the cluster. 

Being a good Cloud Citizen also involves streamlining configuration, deployment and upgrade processes. By integrating these tasks seamlessly, the application can facilitate smooth deployments and upgrades, which will lead to more efficiency and ease of management. 

Join us in this session and embark on a journey to develop a perfect(ish) Cloud Native Java Application with Quarkus. This journey will encompass a comprehensive understanding of both the development and operational perspectives, aiming to optimize applications for the cloud environment. You will get an overview of the Kubernetes basics from a Java developer’s perspective and learn step-by-step how to get your application production ready on Kubernetes.

**[Quarkus Community BOF - Devoxx.be Edition](https://devoxx.be/talk/?id=2945)**
**Session Type:** BOF
**Speaker(s):** Dimitris Andreadis
**Day/Time:** Thursday from 18:50 - 19:50
**Abstract:** Quarkus, the Black Swan of Java, is getting more and more popularity for its unbeatable performance characteristics and its first rate Developer experience. With more than 700+ contributors to the main project, a rapidly expanding ecosystem with 500+ extensions, and a continuous stream of innovation, Quarkus is leading the way in the Kubernetes Native Java Application development space.

Whether you are a current Quarkus user, a future one, a contributor or just a curious mind, you are very much welcome to join us in this session with distinguished members of the Quarkus Development Team that are present at Devoxx.be. Help us shape its future by bringing your ideas about connecting Quarkus and AI or extending to the EDGE, discuss any pain points or the latest around Quarkus v3, find out how you can contribute, and pretty much Ask Us Anything.

It's also pretty much guaranteed that Stephan's GhatGPT generated schedule will assign the worst possible slot for this session, so to reward you, we'll have some nice stickers to give away 🙂

See you there!

**Comment by dvsingh9** [comment]
Thanks for sharing. See you on Monday
